core/src/main/java/org/infinispan/interceptors/locking/AbstractLockingInterceptor.java

```diff
@@ -96,14 +95,15 @@ protected Object visitNonTxDataWriteCommand(InvocationContext ctx, DataWriteComm
    @Override
    public final Object visitInvalidateCommand(InvocationContext ctx, InvalidateCommand command) throws Throwable {
       try {
-         boolean skipLocking = hasSkipLocking(command);
-         long lockTimeout = getLockAcquisitionTimeout(command, skipLocking);
-         for (Object key : command.getKeys()) {
-            lockKey(ctx, key, lockTimeout, skipLocking);
+         if (hasSkipLocking(command)) {
+            return invokeNextInterceptor(ctx, command);
          }
+         lockAllAndRecord(ctx, Arrays.stream(command.getKeys()), getLockTimeoutMillis(command));
```

Comment:
@wburns recommended to use a Stream to avoid creating unnecessary collections to throw away after the locking. 
https://github.com/infinispan/infinispan/pull/3594#discussion_r35098510